clang/docs/ThreadSafetyAnalysis.rst

```diff
@@ -515,7 +515,8 @@ Warning flags
   + ``-Wthread-safety-analysis``: The core analysis.
   + ``-Wthread-safety-precise``: Requires that mutex expressions match precisely.
        This warning can be disabled for code which has a lot of aliases.
-  + ``-Wthread-safety-reference``: Checks when guarded members are passed by reference.
+  + ``-Wthread-safety-reference``: Checks when guarded members are passed or
+    returned by reference.
```

Comment:
Technically, returning is covered by a subflag `-Wthread-safety-reference-return` which we apparently forget to document. On the other hand, we might also get rid of that at some point, it was mainly introduced to ease migration, not because we had evidence that users might want a warning on passing but not on returning.